### PR TITLE
test(DM): add check log contain to `safe-mode` integration test

### LINKED ISSUE #7099: unstable integration test dm/safemode
### Which jobs are flaking?

https://ci2.pingcap.net/blue/organizations/jenkins/dm_ghpr_integration_test/detail/dm_ghpr_integration_test/9592/pipeline

### Which test(s) are flaking?

dm/integration/safemode

### Jenkins logs or GitHub Actions link

```log
(paste logs here)
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: ref #4159, close #7099

### What is changed and how it works?

- the worker1 didn't enter sync status and execute dml, after check_diff before kill_worker, so its SafeModeExitPoint is yet same as its GlobalCheckpoint(beginPosition). This is the unstable reason.

#### Tests 

 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: dm/tests/safe_mode/run.sh [hunk 1]
```diff
@@ -206,6 +206,9 @@ function safe_mode_duration() {
 	dmctl_operate_source create $WORK_DIR/source2.yaml $SOURCE_ID2
 
 	dmctl_start_task "$cur/conf/dm-task-safe-mode-duration.yaml" "--remove-meta"
+	# Worker1 need to start relay, so it is slow.
+	# It is possible that worker1 has not loaded before check diff.
+	sleep 3
```
Comment (GMHDBJD) [inline dm/tests/safe_mode/run.sh:211]: I don't think sleep can make it stable, you may use check_log_contains_with_retry to make sure test enter a centain state.

File: dm/tests/safe_mode/run.sh [hunk 2]
```diff
@@ -211,6 +211,8 @@ function safe_mode_duration() {
 	run_sql_file $cur/data/db1.increment.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
 	run_sql_file $cur/data/db2.increment.sql $MYSQL_HOST2 $MYSQL_PORT2 $MYSQL_PASSWORD2
 
+	# make sure worker1 enter to sync status, and complete one dml
+	check_log_contain_with_retry "alter table t1 add column age int" $WORK_DIR/worker1/log/dm-worker.log
```
Comment (lonng) [inline dm/tests/safe_mode/run.sh:215]: The code change doesn't match the PR title. So which one should be changed?

Comment (okJiang) [reply dm/tests/safe_mode/run.sh:215]: it is working in progress. I will update it

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by buchuitoudegou — APPROVED** [review]

**Pull request merged**